File: mvvm-test/src/main/java/com/martin/mvvm/viewmodels/UserViewModel.java
package com.martin.mvvm.viewmodels;

import android.app.Application;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.martin.mvvm.base.BaseViewModel;
import com.martin.mvvm.db.bean.User;
import com.martin.mvvm.repo.UserRepository;

import java.util.List;

public class UserViewModel extends BaseViewModel {
    private final UserRepository repository;
    private final LiveData<List<User>> users;
    private final MutableLiveData<Boolean> showAddUserDialog = new MutableLiveData<>();

    public LiveData<Boolean> getShowAddUserDialog() {
        return showAddUserDialog;
    }

    public UserViewModel(Application application) {
        super(application);
        repository = new UserRepository(application);
        users = repository.getAllUsers();
    }

    public LiveData<List<User>> getUsers() {
        return users;
    }

    public void insert(User user) {
        repository.insert(user);
        showAddUserDialog.setValue(false);
    }

    public void cancelInsert() {
        if (Boolean.TRUE.equals(showAddUserDialog.getValue())) {
            showAddUserDialog.setValue(false);
        }
    }

    public void insertUser() {
        showAddUserDialog.setValue(true);
    }

    public void delete(User user) {
        repository.delete(user);
    }

    public void update(User user) {
        repository.update(user);
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        repository.shutdown();
    }
}

File: room-livedata-recyclerview/src/main/java/com/example/rlr/UserViewModel.kt
package com.example.rlr

import android.app.Application
import android.util.Log
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class UserViewModel(application: Application) : AndroidViewModel(application) {
    var repository: UserRepository = UserRepository(application)
    private val allUsers: LiveData<List<User?>?>? = repository.getUsers()

    fun getUserList(): LiveData<List<User?>?>? {
        return allUsers
    }

    fun insert(user : User) {
        Log.d("UserViewModel", "insert user: $user")
        repository.insert(user)
    }
}
